compiler/AST/ForLoop.cpp

```diff
@@ -635,3 +654,7 @@ Expr* ForLoop::getNextExpr(Expr* expr)
 
   return retval;
 }
+
+bool ForLoop::isInductionVar(Symbol* sym) {
+  return sym == mIndex->symbol();
+}
```

Comment:
I am guessing you were able to get by with this because:

```chpl
for (i,j) in My2DThing
```

`(i,j)` is packed in a tuple. That sound right?

(This is one of the things that really bothers me. I fixed it for foralls in the past, but I had to scrap my branch that did it for `forall`s)

Reply:
Yeah, if you did `foreach (i,j) in ..2D thing..` you'd see `mIndex->symbol` is a tuple (specifcally named `_indexOfInterest`).

Though looking at the AST it looks like we depack it into `i` and `j` and `i` and `j` get flagged as index variables.

So I'm not really sure if what I'm doing here is the right thing or not. But I tried running this example with a `forall` loop and it also only has that packed tuple on its list of induction variables, so that gives me some confidence.

Practically speaking I think what this is used for is to ensure we don't give an implicit intent to the loops index variable, so I'll try and keep this in mind when I start looking at implicit intents. I'm thinking maybe in the tuple case `i` and `j` are def'd in the loop body itself so that ends up not being a concern?

Anyway, let me know if you're comfortable with me leaving this "as is" for the moment or if you want me to dig deeper before merging.